tests/e2e/vm_service_utils.go

```diff
@@ -0,0 +1,586 @@
+	gomega.Expect(statusCode).Should(gomega.BeNumerically("==", 204))
+	framework.Logf("Successfully Deleted namepsace %v in SVC.", namespace)
+}
+
+// getSvcId fetches the ID of the Supervisor cluster
+func getSvcId(vcRestSessionId string) string {
+	vcIp := e2eVSphere.Config.Global.VCenterHostname
+	svcIdFetchUrl := "https://" + vcIp + "/api/vcenter/namespace-management/supervisors/summaries"
+
+	resp := invokeRestAPI4GetRequests(vcRestSessionId, svcIdFetchUrl)
+
+	var v map[string]interface{}
+	gomega.Expect(json.Unmarshal(resp, &v)).NotTo(gomega.HaveOccurred())
+	framework.Logf("Supervisor summary: %v", v)
+	return v["items"].([]interface{})[0].(map[string]interface{})["supervisor"].(string)
+}
+
+// createAndOrGetContentlibId4Url fetches ID of a content lib that matches the given URL, if none are found it creates a
+// new content lib with the given URL and returns its ID
+func createAndOrGetContentlibId4Url(vcRestSessionId string, contentLibUrl string, dsMoId string,
+	sslThumbPrint string) string {
+
+	r := rand.New(rand.NewSource(time.Now().Unix()))
+	contentlibName := fmt.Sprintf("csi-vmsvc-%v", r.Intn(10000))
+	contentLibId, err := getContentLibId4Url(vcRestSessionId, contentLibUrl)
+	if err == nil {
+		gomega.Expect(contentLibId).NotTo(gomega.BeEmpty())
+		return contentLibId
+	}
+
+	vcIp := e2eVSphere.Config.Global.VCenterHostname
+	contentlbCreationUrl := "https://" + vcIp + "/api/content/subscribed-library"
+	reqBody := fmt.Sprintf(`{
+		"name": "%s",
+		"storage_backings": [{
+			"datastore_id": "%s",
+			"type": "DATASTORE"
+		}],
+		"subscription_info": {
+			"authentication_method": "NONE",
+			"automatic_sync_enabled": true,
+			"on_demand": true,
+			"subscription_url": "%s",
+			"ssl_thumbprint": "%s"
+		},
+		"type": "SUBSCRIBED"
+	}`, contentlibName, dsMoId, contentLibUrl, sslThumbPrint)
+
+	resp, statusCode := invokeRestAPI4PostRequests(vcRestSessionId, contentlbCreationUrl, reqBody)
+	gomega.Expect(statusCode).Should(gomega.BeNumerically("==", 201))
+
+	gomega.Expect(json.Unmarshal(resp, &contentLibId)).NotTo(gomega.HaveOccurred())
+	gomega.Expect(contentLibId).NotTo(gomega.BeEmpty())
+	framework.Logf("Successfully created content library 'csi-vmsvc-test-content-lib' for the test: %v", contentLibId)
+
+	return contentLibId
+}
+
+// getContentLibId4Url fetches ID of a content lib that matches the given URL
+func getContentLibId4Url(vcRestSessionId string, url string) (string, error) {
+	var libId string
+	libIds := getAllContentLibIds(vcRestSessionId)
+	for _, libId := range libIds {
+		lib := getContentLib(vcRestSessionId, libId)
+		if lib.url == url {
+			return libId, nil
+		}
+	}
+	return libId, fmt.Errorf("couldn't find a content lib with subscription url '%v'", url)
+}
+
+// getAllContentLibIds fetches IDs of all content libs
+func getAllContentLibIds(vcRestSessionId string) []string {
+	vcIp := e2eVSphere.Config.Global.VCenterHostname
+	contentLibsFetchUrl := "https://" + vcIp + "/api/content/subscribed-library"
+
```

Comment:
what if there is no content library in the setup? We should return an error for that

Reply:
trying to understand why we need to return an error if there are no libs